src/lib/controllib/block/BlockParam.cpp

```diff
@@ -103,6 +139,10 @@ BlockParam<int32_t &>::BlockParam(Block *block, const char *name, bool parent_pr
 	_val(extern_val)
 {
 	update();
+
+	if (param_type(_handle) != PARAM_TYPE_INT32) {
```

Comment:
I'd like to have this enabled only in debug builds. It adds unnecessary overhead most of the time.

Reply:
It's only in the constructor.

Reply:
It still adds to the bootup time, as it is (or will be) used several 100 times.
Plus we have that check in `param_get` already.
I want to avoid adding bloat that will later be hard to find and optimize for. I prefer to have a CHECKED_BUILD target, that we can run on the automated HW testing boards.

Reply:
Removed.